### ISSUE TITLE
Region of Interest rendering

### ISSUE THREAD

**Issue by arpit15** [BachiLi/redner#116]
I am looking to perform optimization for material properties within a Region of Interest(ROI) of the image. Is there a way to pass ROI in the camera model of the redner? 

For ex: camera renders [2048,2048] img and loss function is defined using image of a rectangle from [1000, 500] to [2000, 1000]

**Comment by kfirshemtov** [comment]
You can express it in the loss function:

( mask * (RenderedImg - targetImg))**2

The RenderedImg is defined according to the resolution of the image you want and the mask is binary indicator for your ROI.

Does it what you mean?

**Comment by arpit15** [comment]
If I render the full-size image, it will take a long time. Therefore I was wondering if there a way to only choose the pixel area to sample from.

**Comment by BachiLi** [comment]
This is probably not difficult to implement. In my TODO list now.

**Comment by BachiLi** [comment]
Added this in 0.4.2.
Now there is a `viewport` argument in the `Camera` class. It's an optional size-4 Tuple, representing (left_top_y, left_top_x, bottom_right_y, bottom_right_x). For example, (0, 0, height, width) represents the full viewport. Set the viewport like this:
```
cam = pyredner.Camera(position = torch.tensor([0.0, 0.0, -5.0]),
                      look_at = torch.tensor([0.0, 0.0, 0.0]),
                      up = torch.tensor([0.0, 1.0, 0.0]),
                      fov = torch.tensor([45.0]),
                      resolution = (1024, 1024),
                      viewport = (200, 300, 700, 800))
```

**Comment by arpit15** [comment]
Thanks!
